spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/AbstractDiscoveryLifecycle.java

```diff
@@ -26,6 +26,8 @@
 import org.springframework.beans.BeansException;
 import org.springframework.boot.context.embedded.EmbeddedServletContainerInitializedEvent;
 import org.springframework.cloud.client.discovery.event.InstanceRegisteredEvent;
+import org.springframework.cloud.client.serviceregistry.Registration;
```

Comment:
In fact, `DiscoveryLifecycle` doesn't even make sense anymore. Maybe `AutoRegistration`. It's not a function of discovery at all.